File: 6_lab_NoPattern/DoctorSpeciality.cs
using System;
using System.Collections.Generic;

namespace _6_lab_NoPattern;

public partial class DoctorSpeciality
{
    public long Id { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public DateOnly DateStart { get; set; }

    public DateOnly? DateEnd { get; set; }

    public long SpecialityId { get; set; }

    public long DoctorId { get; set; }

    public long DepartmentId { get; set; }

    public virtual Department Department { get; set; } = null!;

    public virtual Doctor Doctor { get; set; } = null!;

    public virtual Speciality Speciality { get; set; } = null!;
}

File: 2_Lab_MongoDb/models/doctor_speciality.js
const mongoose = require('mongoose');

const doctor_specialitySchema = new mongoose.Schema({
    date_start: Date,
    date_end: Date,
    speciality_id:{
        type: mongoose.Schema.Types.ObjectId,
        ref: 'speciality'
    },
    doctor_id:{
        type: mongoose.Schema.Types.ObjectId,
        ref: 'doctor'
    },
    department_id:{
        type: mongoose.Schema.Types.ObjectId,
        ref: 'medical_instituction'
    }
});

module.exports = mongoose.model('doctor_speciality', doctor_specialitySchema);